force-app/main/default/classes/WeatherService.cls

```
@@ -0,0 +1,47 @@
+global class WeatherService {
+
+    private static final String META_DATA_LABEL = 'openWeather';
+    private static final String ENDPOINT = 'callout:WeatherAPI?q={0}&units=metric&appid={1}';
+
+    @AuraEnabled(cacheable=true)
+    public static WeatherData getCurrentWeather(String location){ 
+        if(location == null || location ==''){
+            return null;
+        }
+        String apiKey = ([SELECT Key__c FROM WeatherAPIKey__mdt WHERE Label =: META_DATA_LABEL] == null) ? null : [SELECT Key__c FROM WeatherAPIKey__mdt WHERE Label =: META_DATA_LABEL].Key__c;
+        if (apiKey == null || apiKey == '') {
+            return null;
+        }
+        HttpRequest req = new HttpRequest();
+        List<String> endpointValues = new List<String>{location, apiKey};
+        req.setEndpoint(String.format(ENDPOINT, endpointValues));
+        req.setMethod('GET');
+        Http http = new Http();
+        HttpResponse resp = http.send(req);
+        if (resp == null) {
+            return null;
```

Silent fail. Better to throw exception.

---

Fixed